man/fi_endpoint.3.md

```diff
@@ -714,50 +792,186 @@ order values are:
+which message data is sent or received by the transport layer.  Message
+ordering requires matching ordering semantics on the transmitting side of a
+data transfer operation in order to guarantee that ordering is met.
+
+*FI_ORDER_NONE*
+: No ordering is specified.  This value may be used as input in order
+  to obtain the default message order supported by the provider.
+
+*FI_ORDER_RAR*
+: Read after read.  If set, RMA and atomic read operations are
+  processed in the order that they were transmitted relative to other
+  RMA and atomic read operations.  If not set, RMA and atomic reads
+  may be processed out of order from their submission.
+
+*FI_ORDER_RAW*
+: Read after write.  If set, RMA and atomic read operations are
+  processed in the order that they were transmitted relative to RMA
+  and atomic write operations.  If not set, RMA and atomic reads may be
+  processed ahead RMA and atomic writes.
+
+*FI_ORDER_RAS*
+: Read after send.  If set, RMA and atomic read operations are
+  processed in the order that they were transmitted relative to message send
+  operations, including tagged sends.  If not set, RMA and atomic
+  reads may be processed ahead of sends.
+
+*FI_ORDER_WAR*
+: Write after read.  If set, RMA and atomic write operations are
+  processed in the order that they were transmitted relative to RMA and atomic
+  read operations.  If not set, RMA and atomic writes may be processed
+  ahead of RMA and atomic reads.
+
+*FI_ORDER_WAW*
+: Write after write.  If set, RMA and atomic write operations are
+  processed in the order that they were transmitted relative to other RMA
+  and atomic write operations.  If not set, RMA and atomic writes may be
+  processed out of order from their submission.
+
+*FI_ORDER_WAS*
+: Write after send.  If set, RMA and atomic write operations are
+  processed in the order that they were transmitted relative to message send
+  operations, including tagged sends.  If not set, RMA and atomic
+  writes may be processed ahead of sends.
+
+*FI_ORDER_SAR*
+: Send after read.  If set, message send operations, including tagged
+  sends, are processed in the order that they were transmitted relative to RMA
+  and atomic read operations.  If not set, message sends may be processed ahead
+  of RMA and atomic reads.
+
+*FI_ORDER_SAW*
+: Send after write.  If set, message send operations, including tagged
+  sends, are processed in the order that they were transmitted relative to RMA
+  and atomic write operations.  If not set, message sends may be processed ahead
+  of RMA and atomic writes.
+
+*FI_ORDER_SAS*
+: Send after send.  If set, message send operations, including tagged
+  sends, are processed in the order that they were transmitted relative to other
+  message send.  If not set, message sends may be processed out of
+  order from their submission.
+
+## comp_order - Completion Ordering
+
+For a description of message ordering, see the msg_order field in
+the _Transmit Context Attribute_ section.
```

Comment:
Minor nit: I think this should say completion ordering and `comp_order` instead.